Address several Deprecated messages

MrPetovan opened the pull request:
- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783
- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1340019523
- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1336478368
- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1342617099

MrPetovan committed:
Replace parse_url with UriInterface instantiation in Model\APContact

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Coalesce null activity name in ActivityPub\Processor

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Check for parts key existence before performing strpos in Network\Probe

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Provide default host value to hash() in Model\Item::guidFromUri

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Ensure strpos is provided a string in App::runFrontend

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Ward against invalid name from receiver in ActivityPub::storeReceivers

- Address part of https://github.com/friendica/friendica/issues/12011#issuecomment-1338133783

MrPetovan committed:
Check for last-activity value before feeding it to strtotime in Module\NoScrape

- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1340019523

MrPetovan committed:
Ward against empty JsonLD::fetchElement result in ActivityPub\Receiver::processTags

- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1336478368

annando commented on src/Contact/Avatar.php:
```diff
@@ -125,7 +125,7 @@ public static function storeAvatarByImage(array $contact, Image $image): array
 
 	private static function getFilename(string $url): string
 	{
-		$guid = Item::guidFromUri($url, parse_url($url, PHP_URL_HOST));
```
As far as I see the second parameter is used in that function.

annando commented on src/Model/APContact.php:
```diff
@@ -291,14 +291,11 @@ public static function getByURL(string $url, $update = null): array
-				$apcontact['addr'] = $apcontact['nick'] . '@' . str_replace('//', '', Network::unparseURL($parts));
-			} else {
+			try {
+				$apcontact['addr'] = $apcontact['nick'] . '@' . (new Uri($apcontact['url']))->getAuthority();
```
What exactly is `getAuthority` returning? The replaced method took the port number into account. So server at port 10000 had been: `user@domain.tld:10000`.

MrPetovan replied:
And this would be what the original snippet would have returned as well since it was only stripping the scheme and the path. But I'm glad you were able to know exactly what is returned now thanks to the use of the UriInterface object.

annando replied:
Well, with the return value of `getAuthority` I'm slightly helpless, since the name of the function doesn't ring a bell. But a look in the replaced coding told be what was expected. So this other function is doing the same?

MrPetovan replied:
URL authority is `[user[:password]@]domain.tld[:port]`.

annando commented on src/Model/Contact.php:
```diff
@@ -2671,7 +2671,7 @@ private static function updateFromProbeArray(int $id, array $ret): bool
 		}
 
 		$update = false;
-		$guid = ($ret['guid'] ?? '') ?: Item::guidFromUri($ret['url'], parse_url($ret['url'], PHP_URL_HOST));
```
Same question like above: Why is it removed?

annando commented on src/Model/Item.php:
```diff
@@ -2046,6 +2046,8 @@ public static function guidFromUri(string $uri, string $host = null): string
 		// Remove the scheme to make sure that "https" and "http" doesn't make a difference
 		unset($parsed['scheme']);
 
+		$host = $host ?? $parse['host'] ?? 'example.com';
```
`example.com`?

MrPetovan replied:
Why not? When the host was missing before it would have been using empty string which is less fun.

annando replied:
I would prefer some logging with all parameters and the callstack, since there *should* be a resolvable host name provided at any time and just use `''` when the host was not provided.

MrPetovan replied:
Ok. I still stand by the fact that it's less fun.

annando replied:
Yeah, it is, sorry for being German :grin:

MrPetovan committed:
Remove superfluous strlen call in Module\Tos

- Address https://github.com/friendica/friendica/issues/12011#issuecomment-1342617099

annando commented on src/Model/Item.php:
```diff
@@ -2047,11 +2048,16 @@ public static function guidFromUri(string $uri, string $host = null): string
 
+		$hostPart = $host ?? $parsed['host'] ?? '';
+		if (!$hostPart) {
+			Logger::warning('Empty host GUID part', ['uri' => $uri, 'host' => $host, 'parsed' => $parsed]);
```
Please also add the callstack.

MrPetovan replied:
Done.
